### DISCUSSION TITLE
bindBean but for java records

### DISCUSSION THREAD

**Discussion by sombriks** [jdbi/jdbi#2697] [General]
i am wondering if there is a way to bind parameters from a java record.

in order to map results i got it working fine and nice using `jdbi.registerRowMapper(ConstructorMapper.factory(Board.class));` but no luck so far with named parameters and records.

any strategy besides use pojos or something like that?

**Comment by stevenschlansker** [comment]
Hi @sombriks , what did you try so far? Did you see `@BindMethods` which I think should just work for you?

**Reply by sombriks** [reply]
Hi, Thanks for the guidance, indeed bindMethods works perfectly with records, i just used the wrong alias name last time i tried.

```java
// ...
  public Board find(Board board) {
    return database.jdbi.withHandle(handle -> handle.createQuery("""
                select * from boards
                 where id = :id and game_maps_id = :mapId
            """).bindMethods(board)
        .mapTo(Board.class).one());
  }
//...
```

it is actually pretty nead, simple to map from and to a record.

**Reply by stevenschlansker** [reply]
Perfect - we like making it easy :)
